samples/listctrl/listtest.h

```diff
@@ -83,11 +86,15 @@ class MyListCtrl: public wxListCtrl
     void LogColEvent(const wxListEvent& event, const wxString& eventName);
 
     virtual wxString OnGetItemText(long item, long column) const wxOVERRIDE;
+    virtual bool OnGetItemIsChecked(long item) const wxOVERRIDE;
     virtual int OnGetItemColumnImage(long item, long column) const wxOVERRIDE;
     virtual wxItemAttr *OnGetItemAttr(long item) const wxOVERRIDE;
 
     long m_updated;
 
+    // checked boxes in virtual list
+    WX_DECLARE_HASH_MAP(long, bool, wxIntegerHash, wxIntegerEqual, wxLongBoolMap);
```

Comment:
Could also use `wxSelectionStore` here: in spite of its name, it should be equally suitable for storing checked state as selected one.